### PR TITLE
[WIP] Qbert PPO example script

### PR DESCRIPTION
**Pull request by yngtodd**
I thought that a good way to familiarize myself with the library would be to create an example script for the PPO algorithm based on this [yaml](https://github.com/MillionIntegrals/vel/blob/master/examples-configs/rl/atari/ppo/qbert_ppo.yaml) config.

I am not quite sure how to add the `cliprange` parameter to the `vel.rl.algo.policy_gradient.ppo` bit of the algorithm. Currently this example breaks down at that point. I'll add the error message below.

### CODE REVIEW

File: examples-scripts/rl/atari/ppo/qbert_ppo.py [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+            batch_size=256,
+            experience_replay=4
+        ),
+        model=model,
+        algo=PpoPolicyGradient(
+            entropy_coefficient=0.01,
+            value_coefficient=0.5,
+            max_grad_norm=0.5,
+            cliprange=cliprange
+        ),
```
Comment (yngtodd) [inline examples-scripts/rl/atari/ppo/qbert_ppo.py]: Is this how you would add a `LinearSchedule` to `PpoPolicyGradient`?

**Comment by yngtodd** [comment]
Currently the example gets caught when checking the progress in the `LinearSchedule` for `cliprange`:

```
Traceback (most recent call last):
  File "qbert_ppo.py", line 109, in <module>
    qbert_ppo()
  File "qbert_ppo.py", line 103, in qbert_ppo
    reinforcer.train_epoch(epoch_info)
  File "/home/ygx/dev/nykyinen/vel/vel/rl/reinforcers/on_policy_iteration_reinforcer.py", line 84, in train_epoch
    self.train_batch(batch_info)
  File "/home/ygx/dev/nykyinen/vel/vel/rl/reinforcers/on_policy_iteration_reinforcer.py", line 138, in train_batch
    rollout=batch_rollout
  File "/home/ygx/dev/nykyinen/vel/vel/rl/api/base/algo_base.py", line 50, in optimizer_step
    batch_result = self.calculate_gradient(batch_info=batch_info, device=device, model=model, rollout=rollout)
  File "/home/ygx/dev/nykyinen/vel/vel/rl/algo/policy_gradient/ppo.py", line 41, in calculate_gradient
    current_cliprange = self.cliprange.value(batch_info['progress'])
  File "/home/ygx/dev/nykyinen/vel/vel/api/info.py", line 333, in __getitem__
    return self.data_dict[item]
KeyError: 'progress'
```
Should I be passing a `progress_indicator` here?

**Comment by MillionIntegrals** [comment]
Thank you for your interest once again. My apologies for it not being entirely clear, but yeah - lots of pieces need to work there together.

To make this example script to work you need a few additions. It is being taken care of when used with`RlTrainCommand` but when a handrolled training loop is being used (like in this example) it needs to be done manually.

When a cliprange is using a schedule - that means it's value is interpolated linearly from 0.1 to 0.0 during the training. To track that, we need some kind of internal counter of progress - which frame are we currently processing out of how many in total. 

For that there is a special callback that needs to be included in the example.

Import it like that:

```python
from vel.rl.commands.rl_train_command import FrameTracker
```

Then add this callback to the `TrainingInfo` object together with some extra information how many frames do we plan full training to use:

```python
    # Overall information store for training information
    training_info = TrainingInfo(
        metrics=[
            EpisodeRewardMetric('episode_rewards'),  # Calculate average reward from episode
        ],
        callbacks=[
            StdoutStreaming(),  # Print live metrics every epoch to standard output
            FrameTracker()  # We need frame tracker to track the progress of learning
        ]
    )

    training_info['total_frames'] = 1.1e7  # How many frames in the whole training process
```

With these changes I've managed to run this example.

**Comment by yngtodd** [comment]
Great, thanks! With that change everything runs on my end too. I think this should now line up with the yaml config I mentioned above. 

Thanks again for your help! I'm not sure if you want to keep this as another example script, feel free to close this with or without pulling it in!

**Pull request merged**